internal/services/devtestlabs/dev_test_lab_resource.go

```diff
@@ -184,7 +186,16 @@ func resourceDevTestLabRead(d *pluginsdk.ResourceData, meta interface{}) error {
 		d.Set("artifacts_storage_account_id", props.ArtifactsStorageAccount)
 		d.Set("default_storage_account_id", props.DefaultStorageAccount)
 		d.Set("default_premium_storage_account_id", props.DefaultPremiumStorageAccount)
-		d.Set("key_vault_id", props.VaultName)
+
+		// TODO: remove the following workaround until the issue(https://github.com/Azure/azure-rest-api-specs/issues/17422) is fixed.
+		// The lowercase "resourcegroups" and "microsoft.keyvault" in key value id are returned by Labs_Get API.
+		// This will cause the azurerm_dev_test_lab resource which referencing the key vault id to be re-created after running terraform apply command even though nothing has changed.
+		// Hence, re-build the key vault id as a workaround in terraform.
```

Comment:
This is the expected way to handle resource IDs among the provider - rather than a workaround. Therefore, we might remove these comments.

Reply:
thank you, fixed.